mindmeld/models/taggers/lstm.py

```diff
@@ -210,6 +210,7 @@ def extract_features(self, examples, config, resources, y=None, fit=True):
         Returns:
             (sequence_embeddings, encoded_labels, groups): features for the LSTM network
         """
+        del fit  # unused
```

Comment:
@vijay120 is it correct that we aren't using this? Should we make some other change to actually respect the value of fit?

Reply:
The reason why the `fit` param is not being used here is because in traditional ML models, the fit param is used to scale the input data and featurize it. This is not done in the LSTM model since it directly uses the word embeddings as features without the scaling, hence no such fitting is needed. Hence, the param is not needed.

Reply:
We could technically just do this to solve the issue:
```
        if fit:
            # Train time
            self.resources = resources
            padded_y = self._pad_labels(y, DEFAULT_LABEL)
            y_flat = [item for sublist in padded_y for item in sublist]
            encoded_labels_flat = self.label_encoder.fit_transform(y_flat)
            encoded_labels = []
```

Reply:
So we can solve this issue like this:
```
if fit:
            # Train time
            self.resources = resources
            padded_y = self._pad_labels(y, DEFAULT_LABEL)
            y_flat = [item for sublist in padded_y for item in sublist]
            encoded_labels_flat = self.label_encoder.fit_transform(y_flat)
            encoded_labels = []
```

However, we need to update these lines as well:
![Screen Shot 2019-12-12 at 5 14 00 PM](https://user-images.githubusercontent.com/766285/70761664-ce823000-1d02-11ea-99b7-cd65a022df6f.png)

Reply:
For now I'll just add a comment explaining that we use the value of y to decide whether to train or not